### PR TITLE
feat: environment specific component overrides

### PR DESCRIPTION
**Pull request by dhhyi**
## PR Type

[x] Feature
[x] Refactoring (no functional changes, no API changes)
[x] Build-related changes

## What Is the Current Behavior?

The current customization concept for components does not support having multiple brands.
There is Multi-Theme support, but not everything can be handled that way.
Using custom components supports being upgradable while only supporting one override per source repository.

## What Is the New Behavior?

- HTML templates and Component Styles can be replaced by Angular CLI environment option using custom webpack build.
- refactored setting production mode to support running Angular CLI with multiple configurations
  - `ng serve -c brand,production`
  - not supported with the previous way as fileReplacements are not merged but also overridden
- also includes #531 - chunk setting optimizations
- uses [purgecss](https://purgecss.com/) for css minification

## Does this PR Introduce a Breaking Change?

[x] Yes
[ ] No

## TODO

- [x] documentation
- [ ] purgecss in webpack config (optional)

### CODE REVIEW

**Review by shauke — CHANGES_REQUESTED** [review]
Without the last commit "feat: adding overrides for a brand" - I assume that was just a temporary showcase commit - the changes are almost fine (see minor comments).

**Comment by shauke** [comment]
Is there anything that would be worth a migration note, e.g. the Breaking Changes? Maybe in regards to the `production` configuration changes?

File: purgecss.config.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+  content: ['dist/browser/index.html', 'dist/browser/*.js'],
+  css: ['dist/browser/*.css'],
+  output: 'dist/browser',
+  safelist: [/(p|m)(l|r|x|y|t|b)?-[0-5]/],
```
[the remark this answers is not in the document]
Comment (dhhyi) [reply purgecss.config.js:5]: yes, utilities for padding and margin

File: purgecss.config.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+};
```
[the remark this answers is not in the document]
Comment (dhhyi) [reply purgecss.config.js:6]: I guess you can decide that when working on that ticket 😉

File: docs/check-dead-links.js [hunk 3]
```diff
@@ -97,8 +97,8 @@ glob('**/*.md')
         link =>
           !link.includes('tacton.com') &&
           !link.includes('repository.intershop.de') &&
-          !link.includes('support.intershop.com') &&
-          !link.includes('docs.intershop.com') &&
+          !link.includes('support.intershop.') &&
+          !link.includes('docs.intershop.') &&
```
[the remark this answers is not in the document]
Comment (dhhyi) [reply docs/check-dead-links.js:101]: Recently support sites are frequently unavailable. This should not give us red builds.

File: purgecss.config.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (shauke) [reply purgecss.config.js:6]: Okay, that is a good idea. 👍

**Review by shauke — APPROVED** [review]

**Review by shauke — APPROVED** [review]

**Pull request merged**